### ISSUE TITLE
Attribute validators not working

### ISSUE THREAD

**Issue by ehannes** [my-codeworks/fortnox-api#27]
Attribute validators are not working as intended. Validations included form an attribute module is overwritten by the validations added in the validator including the module.

For instance, adding the following to `validators/attributes/country_code_spec.rb` will make the test fail:

```
class TestValidator
  extend Fortnox::API::Validator::Base
  include Fortnox::API::Validator::Attribute::CountryCode

  using_validations do
  end
end
```

**Comment by ehannes** [comment]
I'm affraid #28 did not solve our problem entirely.

When trying to get attribute validation work together with another resource, `Customer` in my case, I get strange errors.
```
  3) Fortnox::API::Validator::Customer.validate Customer with country_code set to a nonsense string should include error for country_code
     Failure/Error: it{ is_expected.to include_error_for( updated_model, attribute, error_type ) }
       expected "#<Set: {#<Vanguard::Violation resource=#<Fortnox::API::Model::Customer:0x007f90b70f6988 @saved=false, @new=true, @url=nil, @address1=nil, @address2=nil, @city=nil, @country=nil, @comments=nil, @currency=nil, @cost_center=nil, @country_code=nil, @customer_number=nil, @delivery_address1=nil, @delivery_address2=nil, @delivery_city=nil, @delivery_country=nil, @delivery_country_code=nil, @delivery_fax=nil, @delivery_name=nil, @delivery_phone1=nil, @delivery_phone2=nil, @delivery_zip_code=nil, @email=nil, @email_invoice=nil, @email_invoice_bcc=nil, @email_invoice_cc=nil, @email_offer=nil, @email_offer_bcc=nil, @email_offer_cc=nil, @email_order=nil, @email_order_bcc=nil, @email_order_cc=nil, @fax=nil, @invoice_administration_fee=nil, @invoice_discount=nil, @invoice_freight=nil, @invoice_remark=nil, @name=nil, @organisation_number=nil, @our_reference=nil, @phone1=nil, @phone2=nil, @price_list=nil, @project=nil, @sales_account=nil, @show_price_vat_included=nil, @terms_of_delivery=nil, @terms_of_payment=nil, @type=nil, @vat_number=nil, @vat_type=nil, @visiting_address=nil, @visiting_city=nil, @visiting_country=nil, @visiting_zip_code=nil, @way_of_delivery=nil, @your_reference=nil, @zip_code=nil> rule=#<Vanguard::Rule::Nullary::Attribute::Presence attribute_name=:name matcher=#<Vanguard::Matcher::Unary::NOT operand=Vanguard::Matcher::Nullary::>>>,
                         #<Vanguard::Violation resource=#<Fortnox::API::Model::Customer:0x007f90b9d01528 @saved=false, @new=true, @url=nil, @address1=nil, @address2=nil, @city=nil, @country=nil, @comments=nil, @currency=nil, @cost_center=nil, @country_code=nil, @customer_number=nil, @delivery_address1=nil, @delivery_address2=nil, @delivery_city=nil, @delivery_country=nil, @delivery_country_code=nil, @delivery_fax=nil, @delivery_name=nil, @delivery_phone1=nil, @delivery_phone2=nil, @delivery_zip_code=nil, @email=nil, @email_invoice=nil, @email_invoice_bcc=nil, @email_invoice_cc=nil, @email_offer=nil, @email_offer_bcc=nil, @email_offer_cc=nil, @email_order=nil, @email_order_bcc=nil, @email_order_cc=nil, @fax=nil, @invoice_administration_fee=nil, @invoice_discount=nil, @invoice_freight=nil, @invoice_remark=nil, @name="A customer", @organisation_number=nil, @our_reference=nil, @phone1=nil, @phone2=nil, @price_list=nil, @project=nil, @sales_account=nil, @show_price_vat_included=nil, @terms_of_delivery=nil, @terms_of_payment=nil, @type="PRIVATE123nonsense123", @vat_number=nil, @vat_type=nil, @visiting_address=nil, @visiting_city=nil, @visiting_country=nil, @visiting_zip_code=nil, @way_of_delivery=nil, @your_reference=nil, @zip_code=nil> rule=#<Vanguard::Rule::Guard:0x007f90b7a5d300 @matcher=#<Vanguard::Matcher::Unary::Attribute operand=#<Vanguard::Matcher::Nullary::Identity value=true>>, @operand=#<Vanguard::Rule::Nullary::Attribute::Inclusion attribute_name=:type matcher=#<Vanguard::Matcher::Nullary::Inclusion accepted=["PRIVATE", "COMPANY"]>>, @_memoized_method_cache=#<Memoizable::Memory:0x007f90b7a5d2b0 @memory=#<ThreadSafe::Cache:0x007f90b7a5d288 @backend={:attribute_name=>:type}, @default_proc=nil>, @monitor=#<Monitor:0x007f90b7a5d210 @mon_owner=nil, @mon_count=0, @mon_mutex=#<Mutex:0x007f90b7a5d1c0>>>>>,
                         #<Vanguard::Violation resource=#<Fortnox::API::Model::Customer:0x007f90b9da3918 @saved=false, @new=true, @url=nil, @address1=nil, @address2=nil, @city=nil, @country=nil, @comments=nil, @currency=nil, @cost_center=nil, @country_code=nil, @customer_number=nil, @delivery_address1=nil, @delivery_address2=nil, @delivery_city=nil, @delivery_country=nil, @delivery_country_code=nil, @delivery_fax=nil, @delivery_name=nil, @delivery_phone1=nil, @delivery_phone2=nil, @delivery_zip_code=nil, @email=nil, @email_invoice=nil, @email_invoice_bcc=nil, @email_invoice_cc=nil, @email_offer=nil, @email_offer_bcc=nil, @email_offer_cc=nil, @email_order=nil, @email_order_bcc=nil, @email_order_cc=nil, @fax=nil, @invoice_administration_fee=nil, @invoice_discount=nil, @invoice_freight=nil, @invoice_remark=nil, @name="A customer", @organisation_number=nil, @our_reference=nil, @phone1=nil, @phone2=nil, @price_list=nil, @project=nil, @sales_account=nil, @show_price_vat_included=nil, @terms_of_delivery=nil, @terms_of_payment=nil, @type="PRIVATE123nonsense123", @vat_number=nil, @vat_type=nil, @visiting_address=nil, @visiting_city=nil, @visiting_country=nil, @visiting_zip_code=nil, @way_of_delivery=nil, @your_reference=nil, @zip_code=nil> rule=#<Vanguard::Rule::Guard:0x007f90b7a5d300 @matcher=#<Vanguard::Matcher::Unary::Attribute operand=#<Vanguard::Matcher::Nullary::Identity value=true>>, @operand=#<Vanguard::Rule::Nullary::Attribute::Inclusion attribute_name=:type matcher=#<Vanguard::Matcher::Nullary::Inclusion accepted=["PRIVATE", "COMPANY"]>>, @_memoized_method_cache=#<Memoizable::Memory:0x007f90b7a5d2b0 @memory=#<ThreadSafe::Cache:0x007f90b7a5d288 @backend={:attribute_name=>:type}, @default_proc=nil>, @monitor=#<Monitor:0x007f90b7a5d210 @mon_owner=nil, @mon_count=0, @mon_mutex=#<Mutex:0x007f90b7a5d1c0>>>>>,
                         #<Vanguard::Violation resource=#<Fortnox::API::Model::Customer:0x007f90ba080050 @saved=false, @new=true, @url=nil, @address1=nil, @address2=nil, @city=nil, @country=nil, @comments=nil, @currency=nil, @cost_center=nil, @country_code=nil, @customer_number=nil, @delivery_address1=nil, @delivery_address2=nil, @delivery_city=nil, @delivery_country=nil, @delivery_country_code=nil, @delivery_fax=nil, @delivery_name=nil, @delivery_phone1=nil, @delivery_phone2=nil, @delivery_zip_code=nil, @email=nil, @email_invoice=nil, @email_invoice_bcc=nil, @email_invoice_cc=nil, @email_offer=nil, @email_offer_bcc=nil, @email_offer_cc=nil, @email_order=nil, @email_order_bcc=nil, @email_order_cc=nil, @fax=nil, @invoice_administration_fee=nil, @invoice_discount=nil, @invoice_freight=nil, @invoice_remark=nil, @name="A customer", @organisation_number=nil, @our_reference=nil, @phone1=nil, @phone2=nil, @price_list=nil, @project=nil, @sales_account=nil, @show_price_vat_included=nil, @terms_of_delivery=nil, @terms_of_payment=nil, @type=nil, @vat_number=nil, @vat_type="SEVAT123nonsense123", @visiting_address=nil, @visiting_city=nil, @visiting_country=nil, @visiting_zip_code=nil, @way_of_delivery=nil, @your_reference=nil, @zip_code=nil> rule=#<Vanguard::Rule::Guard:0x007f90b7a4fc78 @matcher=#<Vanguard::Matcher::Unary::Attribute operand=#<Vanguard::Matcher::Nullary::Identity value=true>>, @operand=#<Vanguard::Rule::Nullary::Attribute::Inclusion attribute_name=:vat_type matcher=#<Vanguard::Matcher::Nullary::Inclusion accepted=["SEVAT", "SEREVERSEDVAT", "EUREVERSEDVAT", "EUVAT", "EXPORT"]>>, @_memoized_method_cache=#<Memoizable::Memory:0x007f90b7a4fc28 @memory=#<ThreadSafe::Cache:0x007f90b7a4fc00 @backend={:attribute_name=>:vat_type}, @default_proc=nil>, @monitor=#<Monitor:0x007f90b7a4fbb0 @mon_owner=nil, @mon_count=0, @mon_mutex=#<Mutex:0x007f90b7a4fb60>>>>>,
                         #<Vanguard::Violation resource=#<Fortnox::API::Model::Customer:0x007f90ba13a450 @saved=false, @new=true, @url=nil, @address1=nil, @address2=nil, @city=nil, @country=nil, @comments=nil, @currency=nil, @cost_center=nil, @country_code=nil, @customer_number=nil, @delivery_address1=nil, @delivery_address2=nil, @delivery_city=nil, @delivery_country=nil, @delivery_country_code=nil, @delivery_fax=nil, @delivery_name=nil, @delivery_phone1=nil, @delivery_phone2=nil, @delivery_zip_code=nil, @email=nil, @email_invoice=nil, @email_invoice_bcc=nil, @email_invoice_cc=nil, @email_offer=nil, @email_offer_bcc=nil, @email_offer_cc=nil, @email_order=nil, @email_order_bcc=nil, @email_order_cc=nil, @fax=nil, @invoice_administration_fee=nil, @invoice_discount=nil, @invoice_freight=nil, @invoice_remark=nil, @name="A customer", @organisation_number=nil, @our_reference=nil, @phone1=nil, @phone2=nil, @price_list=nil, @project=nil, @sales_account=nil, @show_price_vat_included=nil, @terms_of_delivery=nil, @terms_of_payment=nil, @type=nil, @vat_number=nil, @vat_type="SEVAT123nonsense123", @visiting_address=nil, @visiting_city=nil, @visiting_country=nil, @visiting_zip_code=nil, @way_of_delivery=nil, @your_reference=nil, @zip_code=nil> rule=#<Vanguard::Rule::Guard:0x007f90b7a4fc78 @matcher=#<Vanguard::Matcher::Unary::Attribute operand=#<Vanguard::Matcher::Nullary::Identity value=true>>, @operand=#<Vanguard::Rule::Nullary::Attribute::Inclusion attribute_name=:vat_type matcher=#<Vanguard::Matcher::Nullary::Inclusion accepted=["SEVAT", "SEREVERSEDVAT", "EUREVERSEDVAT", "EUVAT", "EXPORT"]>>, @_memoized_method_cache=#<Memoizable::Memory:0x007f90b7a4fc28 @memory=#<ThreadSafe::Cache:0x007f90b7a4fc00 @backend={:attribute_name=>:vat_type}, @default_proc=nil>, @monitor=#<Monitor:0x007f90b7a4fbb0 @mon_owner=nil, @mon_count=0, @mon_mutex=#<Mutex:0x007f90b7a4fb60>>>>>}>"
						 to include inclusion error for "country_code"
```

**Comment by ehannes** [comment]
And test is no longer deterministic but returns different `Vanguard::Violation` every time I run it. When running same spec as above, I now got the following error instead:
```
  4) Fortnox::API::Validator::Customer.validate Customer with country_code set to a nonsense string should include error for country_code
     Failure/Error: it{ is_expected.to include_error_for( updated_model, attribute, error_type ) }
       expected "#<Set: {#<Vanguard::Violation resource=#Fortnox::API::Model::Customer:0x007f97b6ae5f10 @saved=false, @new=true, @name=nil, @url=nil, @address1=nil, @address2=nil, @city=nil, @country=nil, @comments=nil, @currency=nil, @cost_center=nil, @country_code=nil, @customer_number=nil, @delivery_address1=nil, @delivery_address2=nil, @delivery_city=nil, @delivery_country=nil, @delivery_country_code=nil, @delivery_fax=nil, @delivery_name=nil, @delivery_phone1=nil, @delivery_phone2=nil, @delivery_zip_code=nil, @email=nil, @email_invoice=nil, @email_invoice_bcc=nil, @email_invoice_cc=nil, @email_offer=nil, @email_offer_bcc=nil, @email_offer_cc=nil, @email_order=nil, @email_order_bcc=nil, @email_order_cc=nil, @fax=nil, @invoice_administration_fee=nil, @invoice_discount=nil, @invoice_freight=nil, @invoice_remark=nil, @organisation_number=nil, @our_reference=nil, @phone1=nil, @phone2=nil, @price_list=nil, @project=nil, @sales_account=nil, @show_price_vat_included=nil, @terms_of_delivery=nil, @terms_of_payment=nil, @type=nil, @vat_number=nil, @vat_type=nil, @visiting_address=nil, @visiting_city=nil, @visiting_country=nil, @visiting_zip_code=nil, @way_of_delivery=nil, @your_reference=nil, @zip_code=nil> rule=#<Vanguard::Rule::Nullary::Attribute::Presence attribute_name=:name matcher=#<Vanguard::Matcher::Unary::NOT operand=Vanguard::Matcher::Nullary::>>>}>" to include inclusion error for "country_code"
```

**Comment by ehannes** [comment]
Fixed in PR #23...

**Comment by ehannes** [comment]
Fixed in #23
